### PR TITLE
Add support for processing TOML

### PR DESCRIPTION
**Pull request by eayus**
Adds support for processing TOML values per a given specification.

Draft PR since untested as of yet.

### CODE REVIEW

**Comment by eayus** [comment]
Ok I think this is ready to go, I've added a few tests for the basic cases.

File: src/Language/TOML/Processing.idr [hunk 1]
```diff
@@ -0,0 +1,92 @@
+-- SPDX-FileCopyrightText: 2022 The toml-idr developers
+--
+-- SPDX-License-Identifier: MPL-2.0
+
+module Language.TOML.Processing
+
+import Data.List.Quantifiers
+import Language.TOML.Value
+
+-- On Idris 0.5.2 release, remove this in favour of using Data.Singleton.
```
Comment (hydrolarus) [inline src/Language/TOML/Processing.idr]: I don't think this is needed anymore, since you removed the `Singleton` code.
```suggestion

```

**Review by hydrolarus — APPROVED** [review]
This is looking great, thank you!

There's one comment that I think is outdated, but I'll apply that patch and then merge this. 👍

File: src/Language/TOML/Processing.idr (hunk 1 above)
Comment (eayus) [reply src/Language/TOML/Processing.idr]: Oh good catch, yeah that comment is no longer needed.

**Pull request merged**